### ISSUE TITLE
Vertical Navigation A11y review

### ISSUE THREAD

**Issue by joshwooding** [jpmorganchase/salt-ds#5416]
Review vertical navigation's accessibility and improve the component/docs.

**Comment by joshwooding** [comment]
Updates have been made to the pr #5411. @jake-costa is going to add the missing a11y documentation.
